### PR TITLE
Floating Menus: Add Font Family

### LINKED ISSUE #10628: UI Simplification: Floating menu: Font family
## Feature Description

Make the font family element in the floating design menu functional.

Design: [Figma](https://www.figma.com/file/dc9xrVlRhQq4KTE9WQzU1R/_Editor---Sprint-Feature-Level-Designs_?node-id=11394%3A457149)
Menus: Text
Optional rendering: Always renders
Notes: Refactor the existing font family dropdown to meet the new style of this one and use the same dropdown in both places. Perhaps the width varies between the two to correspond to the original dropdown input.

## Additional information

text-font-family

### PR DESCRIPTION
**Pull request by BrookeGraham**
## Context

For use in the floating menus we want to refactor the existing font family dropdown so that we can use the same dropdown in both places. The width / height varies between the two use cases. 

## Summary

We split fontPicker and storyFontPicker into two components in #10848 Here we really jut needed to add some styling props since the use case in the side panel and in the floating menu is slightly different.

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. open floating menu by turning on experiment and selecting text element
2. check that you can change font and it stays in sync with side panel. 

### Does this PR have a security-related impact?
no

### Does this PR change what data or activity we track or use?
no

### Does this PR have a legal-related impact?
no

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #10628

### CODE REVIEW

File: packages/story-editor/src/components/panels/design/textStyle/karma/textStyle.other.karma.js [hunk 1]
```diff
@@ -143,6 +143,11 @@ describe('Text Style Panel', () => {
               currentPage: { elements },
             },
           } = await fixture.renderHook(() => useStory());
+
+          if (!elements) {
+            throw new Error('story elements not ready');
+          }
```
Comment (BrookeGraham) [inline packages/story-editor/src/components/panels/design/textStyle/karma/textStyle.other.karma.js:149]: just an extra check in the outermost waitFor to ensure correct repolling. Keep running into timeout errors locally, so hopefully this helps.

**Comment by timarney** [comment]
Hi @BrookeGraham 

Just a heads up ...  I have a PR in flight for #10014 that's also going to be using the FontPicker.  Just wanted to let you so we can stay in sync with what props are needed for which use case.  


#10014 mentions doing some refactoring to the FontPicker

> FontPicker needs to be refactored so that handleFontPickerChange and fontFamily are passed via props instead (so there is no dependency on selectedElements and pushUpdate

For now in the PR #10808 I'm using the FontPicker pretty much `as is` passing in a "suggested" replacement font via [selectedElements](https://github.com/GoogleForCreators/web-stories-wp/pull/10808/files#diff-86b8778ed4e8771f8f5e36a65d4e42300d7e28179e679d634c94bb3de70afd31R183).   

```javascript
<FontPicker
        selectedElements={[suggestedFont]}
        pushUpdate={onPushUpdate}
        highlightStylesOverride={[]}
      />
```

**Comment by swissspidy** [comment]
Thanks for flagging that, Tim!

In #10014 I proposed the following:

> `FontPicker` needs to be refactored so that `handleFontPickerChange` and `fontFamily` are passed via props instead (so there is no dependency on `selectedElements` and `pushUpdate`

It would be good to have a low-level `FontPicker` like this, so that we can use it in #10808.

For the reusable dropdown for the design panel and floating menu bar, we can still extend it like you're doing here. Can call it `StoryFontPicker` or something.

**Comment by BrookeGraham** [comment]
> FontPicker needs to be refactored so that handleFontPickerChange and fontFamily are passed via props instead (so there is no dependency on selectedElements and pushUpdate

Sorry @swissspidy I am trying to wrap my brain around this.  In this refactor where I am grabbing `selectedElements` and `updateSelectedElements` (for the pushUpdate) directly from state.  Are you saying that you would rather these continue to come in as props?  Or ...

> It would be good to have a low-level FontPicker like this, so that we can use it in https://github.com/GoogleForCreators/web-stories-wp/pull/10808.
>
> For the reusable dropdown for the design panel and floating menu bar, we can still extend it like you're doing here. Can call it StoryFontPicker or something.

Should I leave this refactor as is, and rename it specifically for use in the story-editor?

**Comment by swissspidy** [comment]
@BrookeGraham Yes and no. My idea, based on this PR and #10808:

1. Move the existing `FontPicker` component to `packages/story-editor/src/components/fontPicker/index.js` like in this PR
2. This component should receive two props: `onChange` and `currentValue`. So that there is no dependency on `useStory()` or `useRichTextFormatting()`.
3. Then, create a new component (let's call it `StoryFontPicker`) that does what you're doing in this PR. It returns `<FontPicker onChange={pushUpdate} currentValue={fontFamily} />`.

I whipped up a quick POC for what I had in mind: #10848. PTAL. Hope it makes sense.

**Review by BrittanyIRL — APPROVED** [review]
This is great! it works really well and the sizing is consistent.

**Review by maxyinger — APPROVED** [review]

**Pull request merged**